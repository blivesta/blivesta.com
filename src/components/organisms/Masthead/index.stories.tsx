import React from 'react';
// import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Masthead from '.';

storiesOf(`organisms|Masthead`, module).add(`default`, () => (
  <Masthead siteTitle="blivesta" description="description" />
));
