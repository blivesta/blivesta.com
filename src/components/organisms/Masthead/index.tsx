import * as React from 'react';
import { Link } from 'gatsby';
import styled from 'styled-components';

import Logo from '../../atoms/Logo';
import ContentBlock from '../../molecules/ContentBlock';

const H1 = styled.h1`
  font-family: ${props => props.theme.fonts.avantGarde};
  font-weight: 700;
  font-size: ${props => props.theme.fontSizes[7]};
  margin-top: -8px;
  margin-bottom: ${props => props.theme.spaces[1]};
  line-height: 1;

  ${props => props.theme.media.sm`
    font-size: ${props.theme.fontSizes[8]};
  `}
`;

const StyledLink = styled(Link)`
  display: inline-block;
  width: 100%;
`;

const P = styled.p`
  font-family: ${props => props.theme.fonts.en};
  font-size: ${props => props.theme.fontSizes[1]};
  color: ${props => props.theme.colors.baseText[2]};
  margin-bottom: 0;

  ${props => props.theme.media.sm`
    font-size: ${props.theme.fontSizes[3]};
  `}
`;

interface MastheadProps {
  siteTitle: string;
  description: string;
}

const Masthead = ({ siteTitle, description }: MastheadProps) => {
  return (
    <ContentBlock
      logo={
        <StyledLink to="/">
          <Logo />
        </StyledLink>
      }
    >
      <H1>{siteTitle}</H1>
      <P>{description}</P>
    </ContentBlock>
  );
};

export default Masthead;
