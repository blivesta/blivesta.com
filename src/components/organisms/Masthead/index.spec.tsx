import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Masthead from '.';

describe('Masthead', () => {
  it('snapshot', () => {
    const tree = shallow(<Masthead siteTitle="blivesta" description="description" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
