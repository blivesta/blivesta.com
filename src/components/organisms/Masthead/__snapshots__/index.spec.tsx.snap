// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Masthead snapshot 1`] = `
<ContentBlock
  logo={
    <ForwardRef(Styled(Component))
      to="/"
    >
      <Logo />
    </ForwardRef(Styled(Component))>
  }
>
  <styled.h1>
    blivesta
  </styled.h1>
  <styled.p>
    description
  </styled.p>
</ContentBlock>
`;
